src/promptflow-devkit/promptflow/_sdk/entities/_flows/flex.py

```diff
@@ -33,9 +34,20 @@ def __init__(
         code = Path(code)
         # entry function name
         self.entry = entry
+        self._flow_dir, self._flow_file_name = resolve_flow_path(path,
+                                                                 check_flow_exist=False,
+                                                                 default_flow_file=FLOW_FLEX_YAML)
         # TODO(2910062): support non-dag flow execution cache
         super().__init__(code=code, path=path, dag=data, content_hash=None, **kwargs)
 
+    @property
+    def name(self) -> str:
+        return self._flow_dir.name
+
+    @property
+    def flow_file_path(self) -> Path:
```

Comment:
Please avoid adding public properties for public entities unless necessary

Reply:
can we make this private?

Reply:
Yes, I have made it a private property instead of function.